backend/stakepoold/rpcclient.go

```diff
@@ -110,70 +113,193 @@ func connectWalletRPC(cfg *config) (*dcrrpcclient.Client, semver, error) {
-	for msa, v := range users {
-		addr, err := dcrutil.DecodeAddress(msa, ctx.params)
-		if err != nil {
-			log.Infof("Could not decode multisig address %v for %v: %v",
-				msa, v.Userid, err)
-			continue
-		}
 
-		promises = append(promises, promise{
-			ctx.walletConnection.StakePoolUserInfoAsync(addr), msa})
+	type promise struct {
+		dcrrpcclient.FutureGetTransactionResult
 	}
+	promises := make([]promise, 0, len(tickets))
 
-	var (
-		ticketcount, usercount int
-	)
+	for _, ticket := range tickets {
+		// lookup ownership of each ticket
+		promises = append(promises, promise{ctx.walletConnection.GetTransactionAsync(ticket)})
+	}
 
 	for _, p := range promises {
-		spui, err := p.Receive()
+		gt, err := p.Receive()
 		if err != nil {
-			log.Errorf("unable to fetch tickets for user %v multisigaddr %v: %v",
-				users[p.msa].Userid, p.msa, err)
+			// All tickets should exist and be able to be looked up
+			log.Warnf("GetTransaction error: %v", err)
 			continue
 		}
-
-		if spui == nil || len(spui.Tickets) <= 0 {
-			continue
-		}
-
-		for _, ticket := range spui.Tickets {
-			if ticket.Status != "live" {
+		for i := range gt.Details {
+			_, ok := userVotingConfig[gt.Details[i].Address]
+			if !ok {
+				log.Warnf("Could not map ticket %v to a user, user %v doesn't exist", gt.TxID, gt.Details[i].Address)
 				continue
 			}
 
-			hash, err := chainhash.NewHashFromStr(ticket.Ticket)
+			hash, err := chainhash.NewHashFromStr(gt.TxID)
 			if err != nil {
 				log.Infof("invalid ticket %v", err)
 				continue
 			}
 
-			userTickets[*hash] = users[p.msa].MultiSigAddress
-			ticketcount++
+			// stake pool mode in dcrwallet works by checking the fee when the
+			// ticket is first seen.  If the fee is too low, the ticket is
+			// placed in the invalid bucket.  Otherwise, the ticket is added
+			// to the wallet.
+			// When stake pool-specific functionality in dcrwallet is removed,
+			// most likely all tickets, whether they pay the correct fee or not,
+			// will end up in live tickets list.
+			// So we need to check if the ticket is in the low fee list first
+			// before doing normal fee processing.
+			_, isManuallyAdded := ctx.manuallyAddedTicketsMSA[*hash]
+			if isManuallyAdded {
+				ticketsManuallyAddedCount++
+				userTickets[*hash] = userVotingConfig[gt.Details[i].Address].MultiSigAddress
+			} else {
+				ticketFeesValid, err := nodeCheckTicketFee(ctx, gt.Hex, gt.BlockHash)
+				if ticketFeesValid {
+					userTickets[*hash] = userVotingConfig[gt.Details[i].Address].MultiSigAddress
+				} else {
+					ticketsIgnoredCount++
+					log.Warnf("ignoring ticket %v for msa %v ticketFeesValid %v err %v",
+						*hash, ctx.userVotingConfig[gt.Details[i].Address].MultiSigAddress, ticketFeesValid, err)
+				}
+			}
+			break
 		}
-		usercount++
 	}
 
-	ticketNoun := pickNoun(ticketcount, "ticket", "tickets")
-	userNoun := pickNoun(usercount, "user", "users")
+	atomic.AddUint64(&ctx.stats.TicketsIgnoredCount, uint64(ticketsIgnoredCount))
```

Comment:
I decided to remove all the stats stuff since it was pretty half-baked and I'm not sure it was very useful in the grand scheme of things.